hw/sensors/bmp5: Add BMP5 driver support

vrahane opened the pull request.

andrzej-kaczmarek commented:
Code style checker prefers to exceed line length if the excessive length is relatively short, as in this case. Simply make changes as noted.

You also need to fix licenses -> update `LICENSE` and `.rat-excludes` files. The dates in licenses seem to be odd though, i.e. 2025-2027...

vrahane commented:
@andrzej-kaczmarek  Thanks, added the license for inclusion and updated rat excludes for exclusion. Also, most of the Bosch licenses have two years copyright. So, I just took that.

vrahane commented:
@andrzej-kaczmarek  Updated.
